### ISSUE TITLE
[sklearn#33942] `test_search_cv_sample_weight_equivalence[estimator0]` fails when built with `-march=armv9-a`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#372]
**Mirrored from [scikit-learn#33942](https://github.com/scikit-learn/scikit-learn/issues/33942)**

Original title: `test_search_cv_sample_weight_equivalence[estimator0]` fails when built with `-march=armv9-a`
Original labels: Bug, Needs Triage
Created: 2026-05-07T13:25:10Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the bug and give evidence about its user-facing impact

When built with `-march=armv9-a`, I get the following test error:
```
[  139s] =================================== FAILURES ===================================
[  139s] _____________ test_search_cv_sample_weight_equivalence[estimator0] _____________
[  139s] [gw1] linux -- Python 3.13.13 /usr/bin/python3.13
[  139s] 
[  139s] estimator = GridSearchCV(estimator=LogisticRegression(), param_grid={'C': [1, 10, 100]})
[  139s] 
[  139s]     @pytest.mark.parametrize(
[  139s]         "estimator",
[  139s]         [
[  139s]             GridSearchCV(estimator=LogisticRegression(), param_grid={"C": [1, 10, 100]}),
[  139s]             RandomizedSearchCV(
[  139s]                 estimator=Ridge(), param_distributions={"alpha": [1, 0.1, 0.01]}
[  139s]             ),
[  139s]         ],
[  139s]     )
[  139s]     def test_search_cv_sample_weight_equivalence(estimator):
[  139s]         estimator_weighted = clone(estimator)
[  139s]         estimator_repeated = clone(estimator)
[  139s]         set_random_state(estimator_weighted, random_state=0)
[  139s]         set_random_state(estimator_repeated, random_state=0)
[  139s]     
[  139s]         rng = np.random.RandomState(42)
[  139s]         n_classes = 3
[  139s]         n_samples_per_group = 30
[  139s]         n_groups = 4
[  139s]         n_samples = n_groups * n_samples_per_group
[  139s]         X = rng.rand(n_samples, n_samples * 2)
[  139s]         y = rng.randint(0, n_classes, size=n_samples)
[  139s]         sw = rng.randint(0, 5, size=n_samples)
[  139s]         # we use groups with LeaveOneGroupOut to ensure that
[  139s]         # the splits are the same in the repeated/weighted datasets
[  139s]         groups = np.tile(np.arange(n_groups), n_samples_per_group)
[  139s]     
[  139s]         X_weighted = X
[  139s]         y_weighted = y
[  139s]         groups_weighted = groups
[  139s]         splits_weighted = list(LeaveOneGroupOut().split(X_weighted, groups=groups_weighted))
[  139s]         estimator_weighted.set_params(cv=splits_weighted)
[  139s]         # repeat samples according to weights
[  139s]         X_repeated = X_weighted.repeat(repeats=sw, axis=0)
[  139s]         y_repeated = y_weighted.repeat(repeats=sw)
[  139s]         groups_repeated = groups_weighted.repeat(repeats=sw)
[  139s]         splits_repeated = list(LeaveOneGroupOut().split(X_repeated, groups=groups_repeated))
[  139s]         estimator_repeated.set_params(cv=splits_repeated)
[  

...(truncated)

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [29](https://github.com/godofecht/flow-scikit/issues/29) in the flow-scikit repository. Further library-specific triage and implementation will happen there.

**Comment by godofecht** [comment]
Moved to [28](https://github.com/godofecht/flow-scikit/issues/28) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
